zebra-chain/src/block/difficulty.rs

```diff
@@ -93,6 +93,45 @@ impl fmt::Debug for ExpandedDifficulty {
     }
 }
 
+/// A 128-bit unsigned "Work" value.
+///
+/// Used to calculate the total work for each chain of blocks.
+///
+/// Details:
+///
+/// The relative value of `Work` is consensus-critical, because it is used to
+/// choose the best chain. But its precise value and bit pattern are not
+/// consensus-critical.
+///
+/// We calculate work values according to the Zcash specification, but store
+/// them as u128, rather than the implied u256. We don't expect the total chain
+/// work to ever exceed 2^128. The current total chain work for Zcash is 2^58,
+/// and Bitcoin adds around 2^91 work per year. (Each extra bit represents twice
+/// as much work.)
+#[derive(Clone, Copy, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Work(u128);
+
+impl fmt::Debug for Work {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let log2_work = f64::from_str(&self.0.to_string()).ok().map(f64::log2);
```

Comment:
I've been informed that this also "just works"
```suggestion
        let log2_work = (self.0 as f64).log2();
```